redisson/src/main/java/org/redisson/config/ConfigSupport.java

```diff
@@ -136,13 +146,15 @@ private String resolveEnvParams(Readable in) {
     }
     
     private String resolveEnvParams(String content) {
-        Pattern pattern = Pattern.compile("\\$\\{(\\w+)\\}");
+        Pattern pattern = Pattern.compile("\\$\\{(\\w+(:-.+)?)\\}");
         Matcher m = pattern.matcher(content);
         while (m.find()) {
-            String s = m.group(1);
-            String v = System.getenv(s);
+            String[] parts = m.group(1).split(":-");
+            String v = env.get(parts[0]);
```

Comment:
Please don't extract invocation of `System.getenv` to separate provider object.

Reply:
It's a good practice for testability. How would you then test the class?

Reply:
@aaabramov  Through integration test. Basically all Redisson tests are integration.

Reply:
@mrniko I can propose testing it with PowerMock, if it is required. 
Otherwise is there any example I can take where I can override env?

Reply:
Because BaseTest has hard-coded configuration:

```java


    public static Config createConfig() {
//        String redisAddress = System.getProperty("redisAddress");
//        if (redisAddress == null) {
//            redisAddress = "127.0.0.1:6379";
//        }
        Config config = new Config();
//        config.setCodec(new MsgPackJacksonCodec());
//        config.useSentinelServers().setMasterName("mymaster").addSentinelAddress("127.0.0.1:26379", "127.0.0.1:26389");
//        config.useClusterServers().addNodeAddress("127.0.0.1:7004", "127.0.0.1:7001", "127.0.0.1:7000");
        config.useSingleServer()
                .setAddress(RedisRunner.getDefaultRedisServerBindAddressAndPort());
//        .setPassword("mypass1");
//        config.useMasterSlaveConnection()
//        .setMasterAddress("127.0.0.1:6379")
//        .addSlaveAddress("127.0.0.1:6399")
//        .addSlaveAddress("127.0.0.1:6389");
        return config;
    }

    public static RedissonClient createInstance() {
        Config config = createConfig();
        return Redisson.create(config);
    }
```

Reply:
@aaabramov 

> Because BaseTest has hard-coded configuration:

Commented out code is not used anymore

Reply:
@aaabramov 

> I can propose testing it with PowerMock, if it is required.

JMockit is used as mocking library